test/txsim/account.go

```diff
@@ -225,11 +227,29 @@ func (am *AccountManager) Submit(ctx context.Context, op Operation) error {
 
 	opts := make([]user.TxOption, 0)
 	if op.GasLimit == 0 {
-		opts = append(opts, user.SetGasLimit(DefaultGasLimit), user.SetFee(defaultFee))
+		if am.gasLimit > 0 {
+			// Use the custom gas limit set on the AccountManager
+			opts = append(opts, user.SetGasLimit(am.gasLimit))
+			
+			// Calculate fee based on gas price
+			var gasPrice float64
+			if am.gasPrice > 0 {
+				gasPrice = am.gasPrice
+			} else {
+				gasPrice = appconsts.DefaultMinGasPrice
+			}
+			opts = append(opts, user.SetFee(uint64(math.Ceil(float64(am.gasLimit)*gasPrice))))
```

Comment:
[question] this seems a bit difficult to reason about. Is it possible to extract the `fee` calculation to _after_ the custom gas limit and gas price have been set. That way this code can remain less indented:

```go
if am.gasLimit > 0 {
    // set custom gas limit
}

if am.gasPrice > 0 {
  // set custom gas price
}

fee := calculateFee(gasLimit, gasPrice)
opts = append(opts, user.SetFee(fee))
```

That way there's only one line where we set the fee instead of two.

Reply:
thank you for the feedback! https://github.com/celestiaorg/celestia-app/pull/4447/commits/36d2c38486f3494a6426ea38fb5a481f7a454153